docs:  improvements (#283)
nuxt/image: docs/content/en/3.providers.md

---
title: Providers
description: "Nuxt Image supports multiple providers for high performances."
category: "Getting Started"
---

## Default provider

Nuxt Image comes with a preconfigured instance of [IPX](/providers/ipx) to provide images transformations based on the [Sharp](https://www.npmjs.com/package/sharp) package. That default provider can be used without any configuration. The module will automatically optimize `<nuxt-img>` or `<nuxt-picture>` sources and accepts all [options](/api/options/), except for modifiers which are specific to image services providers.

> With Universal target, IPX will generate correct images at runtime. With static target, IPX will generate all optimized assets on `nuxt generate` and store them in the `dist` folder.

Reviewer: Would be nice linking to nuxt docs for target

> ## Image Services Providers

Reviewer: Providers are not necessarily a service. Like vercel is integrated cdn. And ipx is replaceable with a custom non-service provider

---
title: Providers
description: "Nuxt Image supports multiple providers for high performances."
category: "Getting Started"
---

## Default provider

Nuxt Image comes with a preconfigured instance of [IPX](/providers/ipx) to provide images transformations based on the [Sharp](https://www.npmjs.com/package/sharp) package. That default provider can be used without any configuration. The module will automatically optimize `<nuxt-img>` or `<nuxt-picture>` sources and accepts all [options](/api/options/), except for modifiers which are specific to image services providers.

> With Universal target, IPX will generate correct images at runtime. With static target, IPX will generate all optimized assets on `nuxt generate` and store them in the `dist` folder.

Reviewer 2: ```suggestion
With a `server` target, IPX will generate images at runtime. With a `static` target, IPX will generate all optimized assets on running `nuxt generate` and store them in the `dist` folder.
```

[Learn more about Nuxt deployment targets](https://nuxtjs.org/docs/2.x/features/deployment-targets)

## Image Providers

> Providers are an integration between the Image module and image providers. Using a specific provider means that your images will be transformed into URLs from the image service or custom provider.

Reviewer 2: ```suggestion
Providers are integrations between the Nuxt Image module and third-party image transformation services (or, as in the case of `ipx`, a self-hosted/generate-time image transformation service). Each provider is responsible for generating correct URLs for that image transformation service.
```

> Providers are integrations between the Nuxt Image module and third-party image transformation services (or, as in the case of `ipx`, a self-hosted/generate-time image transformation service). Each provider is responsible for generating correct URLs for that image transformation service.

Reviewer: "Transformation service" is not covering imo (also why introducing a new term like this?). Providers, while allowing optional optimization and manipulation/transformation/modification, are for example also CDN provider. A provider is basically URL generator (even for ipx/static)

> Nuxt Image has a generic way to work with external providers. Here is a complete list of providers that are supported out-of-the-box and can be easily configured to use.

Reviewer 2: ```suggestion
Nuxt Image can be configured to work with any external image transformation service. Here is a complete list of providers that are supported out-of-the-box and can be easily configured to use:
```
